Content.Server/Atmos/Piping/Unary/EntitySystems/GasVentPumpSystem.cs

```diff
@@ -164,6 +168,48 @@ private void OnGasVentPumpUpdated(EntityUid uid, GasVentPumpComponent vent, ref
             }
         }
 
+        /// <summary>
+        ///     Keeps track of the average rate of change in pressure, and returns `false` if
+        ///     the rate is less than <see cref=GasVentPumpComponent.pressurizationLockout>.
+        ///
+        ///     Because this was written by someone lazy, multiple calls in an atmos tick aren't properly supported.
+        /// </summary>
+        /// <remarks>
+        ///     This method is intended to stop vents from pumping air into rooms which are rapidly losing pressure.
+        /// </remarks>
+        private bool pressurizationLockout(GasVentPumpComponent vent, GasMixture environment)
+        {
+            if (vent.UnderPressureLockout)
+            {
+                vent.windowidx = -1;
+                vent.samples = 0;
+                return false;
+            }
+            vent.windowidx = (vent.windowidx + 1) % vent.maxSamples;
+
+            var oldAverage = vent.averagePressure;
+            if (vent.samples < vent.maxSamples)
+                vent.samples += 1;
+
+            // Calculate average pressure
+            float pressure = 0;
+            vent.measurements[vent.windowidx] = environment.Pressure;
+            for (int i = 0; i < vent.samples; i++)
+                pressure += vent.measurements[i];
+            vent.averagePressure = pressure / vent.samples;
+
+            // Calculate average rate of pressurization
+            float refill = 0;
+            vent.pressurizationRate[vent.windowidx] = vent.averagePressure - oldAverage;
+            for (int i = 0; i < vent.samples; i++)
+                refill += vent.pressurizationRate[i];
+            var averagePressurizationRate = refill / vent.samples;
```

Comment:
This is a little confusing to me. It looks like you want the boxcar-windowed average value (moving average) of the time-derivative of pressure. But instead you're calculating an average pressure, finding the change within the window, and then averaging it again? Why not:

```
dP = currentPressure - lastPressure;
averagePressurizationRate = MovingAverage(dP, 5);
```

It seems to me that doing it this way effectively creates a critically-damped 2-pole system that responds very slowly to input changes.

Reply:
What you are suggesting might very well be better than whatever the cursed abomination I created is.
I ended up with my current code by essentially stacking a bunch of tiny fixes, until I got this thing to work well enough. 
Currently I don't think the slow response time is that bad, though I haven't tested how big of an impact it has on distro.